libraries/scalahadooplibrary/build.gradle

```diff
@@ -0,0 +1,11 @@
+apply plugin: "application"
+apply plugin: "java-library"
+
+
+dependencies {
+    // spark is provided by spark-submit
+    compileOnly deps.external.sparkCore
+    compileOnly deps.external.sparkSql
+
+    implementation deps.external.log4j
```

Comment:
@kageiit -- commenting out this line makes `//libraries/scalahadooplibrary/...` to build successfully. 

For some reason, the compile classpath doesn't contain `log4j` which is why the resolved `deps` list doesn't contain the expected log4j dependency, causing the build to fail.  (`intersection(getMain().getExternalDeps(), getProvided().getExternalDeps())`)

On the other hand when `implementation deps.external.log4j` is removed since log4j shows up in the provided list the compilation is successful.

Reply:
was able to figure out the issue. We do set intersection and had to do versionless set intersection instead to get the right output.